wfepy/workflow.py

```diff
@@ -108,11 +108,13 @@ class Runner:
     :ivar workflow: :class:`Workflow`
     :ivar context: arbitrary user object, passed to all tasks
     :ivar state: state of execution
+    :ivar completed_tasks: completed tasks
     """
 
     workflow = attr.ib()
     context = attr.ib(default=None)
     state = attr.ib(default=None, init=False)
+    completed_tasks = attr.ib(default=[])
```

Comment:
The problem with tests is that the default list is actually used by instances and changes persist for later instances (i.e. the list is not copied). Fix:

```suggestion
    completed_tasks = attr.ib(factory=list)
```

Reply:
Ah. Ok, Thanks @hluk